seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/split/FileSplitUtils.java

```diff
@@ -0,0 +1,219 @@
+        long fileSize = fileStatus.getLen();
+        long splitSizeBytes = splitSizeMB * 1024L * 1024L;
+
+        // If file is smaller than split size, return single split
+        if (fileSize <= splitSizeBytes) {
+            splits.add(new FileSourceSplit(tableId, filePath));
+            return splits;
+        }
+
+        // Calculate split positions
+        List<Long> splitPositions =
+                calculateSplitPositions(filePath, fileSize, splitSizeBytes, hadoopFileSystemProxy);
+
+        // Create splits based on positions
+        for (int i = 0; i < splitPositions.size(); i++) {
+            long startOffset = i == 0 ? 0 : splitPositions.get(i - 1);
+            long length =
+                    (i == splitPositions.size() - 1)
+                            ? (fileSize - startOffset)
+                            : (splitPositions.get(i) - startOffset);
+            boolean isFirstSplit = (i == 0);
+
+            splits.add(new FileSourceSplit(tableId, filePath, startOffset, length, isFirstSplit));
+        }
+
+        log.info(
+                "Split file {} ({} bytes) into {} splits of approximately {} MB each",
+                filePath,
+                fileSize,
+                splits.size(),
+                splitSizeMB);
+
+        return splits;
+    }
+
+    /** Calculate split positions that respect row boundaries. */
+    private static List<Long> calculateSplitPositions(
+            String filePath,
+            long fileSize,
+            long splitSizeBytes,
+            HadoopFileSystemProxy hadoopFileSystemProxy)
+            throws IOException {
+
+        List<Long> positions = new ArrayList<>();
+
+        try (FSDataInputStream inputStream = hadoopFileSystemProxy.getInputStream(filePath)) {
+            long currentPosition = 0;
+
+            while (currentPosition < fileSize) {
+                long targetPosition = Math.min(currentPosition + splitSizeBytes, fileSize);
+
+                if (targetPosition >= fileSize) {
+                    // No more splits needed
+                    break;
+                }
+
+                // Find the next line boundary after targetPosition
+                long actualSplitPosition =
+                        findNextLineBoundary(inputStream, targetPosition, fileSize);
+
+                if (actualSplitPosition > currentPosition) {
+                    positions.add(actualSplitPosition);
+                    currentPosition = actualSplitPosition;
+                } else {
+                    // If we can't find a line boundary, break to avoid infinite loop
+                    break;
+                }
+            }
+        }
+
+        return positions;
+    }
+
+    /** Find the next line boundary after the given position. */
+    private static long findNextLineBoundary(
+            FSDataInputStream inputStream, long startPosition, long fileSize) throws IOException {
+
+        inputStream.seek(startPosition);
```

Comment:
some inputstream unsupport seek operation (For example, ftp). So I think we should precheck before use this new feature.